data_processing/process_data.py

```diff
@@ -0,0 +1,30 @@
+# This script handles the processing and creation of the interaction matrix
+
+import numpy as np
+import os
+
+def create_interaction_matrix(directory_path, num_users, num_movies, files):
+    interaction_matrix = np.zeros((num_users, num_movies), dtype=np.int8)
```

Comment:
Are you using a coo_matrix here? It looks like a normal one. Even if lightFM allows for a non-sparse matrix, we should use a sparse one to save memory.